lib/modules/apostrophe-schemas/index.js

```diff
@@ -977,14 +977,21 @@ module.exports = {
       converters: {
         string: function(req, data, name, object, field, callback) {
           object[name] = self.apos.launder.string(data[name], field.def);
-          if (field.min && (object[name].length < field.min)) {
+          if (object[name] && field.min && (object[name].length < field.min)) {
             // Would be unpleasant, but shouldn't happen since the browser
             // also implements this. We're just checking for naughty scripts
             return callback('min');
           }
-          if (field.max && (object[name].length > field.max)) {
+          // If max is longer than allowed, trim the value down to the max length
+          if (object[name] && field.max && (object[name].length > field.max)) {
             object[name] = object[name].substr(0, field.max);
           }
+          // If field is required but empty (and client side didn't catch that)
+          // This is new and until now if JS client side failed, then it would
+          // allow the save with empty values -Lars
+          if (field.required && _.isEmpty(object[name])) {
```

Comment:
@boutell Corrected to only check `min` and `max` if field has a submitted value.

Additionally, by semi accident, a new validation on `required`. In my testing I noticed that if the JavaScript client side code fails (or is not present), then the `null` value will actually fly all the way through and into the DB. So this is a humble first attempt to validate empty submits for a field that is `required` server side. At this time the UI does show a notification that "an error happened" when the server side validation kicks in, and that message should probably be improved (struggling to understand all the code here), but this is like a last resort if the client side code does not catch it.

My main concerns are:

* Is there any bc issues or bad side effects of adding the `required` check server side? It's surprising me that there was no `required` check server side until now, so I'm thinking there may have been a valid reason for that... I have no problem removing this `required` validation to get the rest of this PR in, but this should probably be addressed sooner rather than later and this at least stops the save and errors out. This was just something I hit by trying to improve the other part and not really related to my original PR.

* Should this type of logic be inside the converters string function? If not, please help guiding me where would be a better home for this. Also struggling with overall full understanding of the entire code base.